test/e2e/storage/testsuites/multivolume.go

```diff
@@ -0,0 +1,468 @@
+			}
+			resource := createGenericVolumeTestResource(driver, l.config, curPattern)
+			l.resources = append(l.resources, resource)
+			pvcs = append(pvcs, resource.pvc)
+		}
+
+		TestAccessMultipleVolumesAcrossPodRecreation(l.config.Framework, l.cs, l.ns.Name,
+			framework.NodeSelection{Name: l.config.ClientNodeName}, pvcs, false /* sameNode */)
+	})
+
+	// This tests below configuration:
+	// [pod1] [pod2]
+	// [   node1   ]
+	//   \      /     <- same volume mode
+	//   [volume1]
+	It("should concurrently access the single volume from pods on the same node", func() {
+		init()
+		defer cleanup()
+
+		numPods := 2
+
+		if !l.driver.GetDriverInfo().Capabilities[CapMultiPODs] {
+			framework.Skipf("Driver %q does not support multiple concurrent pods - skipping", dInfo.Name)
+		}
+
+		// Create volume
+		resource := createGenericVolumeTestResource(l.driver, l.config, pattern)
+		l.resources = append(l.resources, resource)
+
+		// Test access to the volume from pods on different node
+		TestConcurrentAccessToSingleVolume(l.config.Framework, l.cs, l.ns.Name,
+			framework.NodeSelection{Name: l.config.ClientNodeName}, resource.pvc, numPods, true /* sameNode */)
+	})
+
+	// This tests below configuration:
+	//        [pod1] [pod2]
+	// [   node1   ] [   node2   ]
+	//         \      /     <- same volume mode
+	//         [volume1]
+	It("should concurrently access the single volume from pods on different node", func() {
+		init()
+		defer cleanup()
+
+		numPods := 2
+
+		if !l.driver.GetDriverInfo().Capabilities[CapRWX] {
+			framework.Skipf("Driver %s doesn't support %v -- skipping", l.driver.GetDriverInfo().Name, CapRWX)
+		}
+
+		// Check different-node test requirement
+		nodes := framework.GetReadySchedulableNodesOrDie(l.cs)
+		if len(nodes.Items) < numPods {
+			framework.Skipf(fmt.Sprintf("Number of available nodes is less than %d - skipping", numPods))
+		}
+		if l.config.ClientNodeName != "" {
+			framework.Skipf("Driver %q requires to deploy on a specific node - skipping", l.driver.GetDriverInfo().Name)
+		}
+
+		// Create volume
+		resource := createGenericVolumeTestResource(l.driver, l.config, pattern)
+		l.resources = append(l.resources, resource)
+
+		// Test access to the volume from pods on different node
+		TestConcurrentAccessToSingleVolume(l.config.Framework, l.cs, l.ns.Name,
+			framework.NodeSelection{Name: l.config.ClientNodeName}, resource.pvc, numPods, false /* sameNode */)
+	})
+}
+
+// TestAccessMultipleVolumesAcrossPodRecreation tests access to multiple volumes from single pod,
+// then recreate pod on the same or different node depending on requiresSameNode,
+// and recheck access to the volumes from the recreated pod
+func TestAccessMultipleVolumesAcrossPodRecreation(f *framework.Framework, cs clientset.Interface, ns string,
+	node framework.NodeSelection, pvcs []*v1.PersistentVolumeClaim, requiresSameNode bool) {
+	By(fmt.Sprintf("Creating pod on %+v with multiple volumes", node))
```

Comment:
L323 - L340 are repeated. Can we put those into a function?

Reply:
@msau42 

Thank you for good advice, I fixed as suggested.
Also, I found that read test should be done before write test after pod recreation, but previous code doesn't do it properly.
So, I fixed it to work properly.

PTAL